ribs/visualize.py

```diff
@@ -453,3 +453,145 @@ def sliding_boundaries_archive_heatmap(archive,
+        cmap (str, list, matplotlib.colors.Colormap): Colormap to use when
+            plotting intensity. Either the name of a colormap, a list of RGB or
+            RGBA colors (i.e. an Nx3 or Nx4 array), or a colormap object.
+        linewidth (float): Line width for solutions on the plot.
+        alpha (float): Opacity of solutions on graph (you may want to turn this
+            down if there are many solutions that are found to see them)
+        vmin (float): Minimum objective value to use in the plot. If None, the
+            minimum objective value in the archive is used.
+        vmax (float): Maximum objective value to use in the plot. If None, the
+            maximum objective value in the archive is used.
+
+    Raises:
+        ValueError: The bcs provided do not exist in the archive.
+        ValueError: The incorrect number of labels are provided.
+    """
+    # Try getting the colormap early in case it fails.
+    cmap = _retrieve_cmap(cmap)
+
+    #retrieve data from archive
+    archive_data = archive.as_pandas(include_solutions=False)
+
+    #if there is no order specified, plot in increasing numerical order
+    if bc_order is None:
+        cols = [key for key in archive_data.keys() if 'behavior' in key]
+        lower_bounds = archive.lower_bounds
+        upper_bounds = archive.upper_bounds
+
+    #use the requested behaviors (may be less than the original number of bcs)
+    else:
+        #check for errors in specification
+        for bc in bc_order:
+            if bc not in archive_data.keys():
+                raise ValueError(f'Behavior {bc} is not in the archive.')
+        cols = bc_order
+        #find the indices of the requested order
+        bc_indices = [int(x.split('_')[1]) for x in bc_order]
+        lower_bounds = archive.lower_bounds[bc_indices]
+        upper_bounds = archive.upper_bounds[bc_indices]
+
+    if axis_labels is not None and len(axis_labels) != len(cols):
+        raise ValueError(f'Label Mismatch: You have {len(cols)} axes'
+                         'and {len(axis_labels)} labels.')
+
+    host = plt.gca() if ax is None else ax  #which axis to plot on
+    df = archive.as_pandas(include_solutions=False)
+    vmin = np.min(df['objective']) if vmin is None else vmin
+    vmax = np.max(df['objective']) if vmax is None else vmax
+    norm = matplotlib.colors.Normalize(vmin=vmin, vmax=vmax, clip=True)
+    objectives = df['objective'].to_numpy()
+    ys = df[cols].to_numpy()
+    y_ranges = upper_bounds - lower_bounds
+
+    # transform all data to be in the first axis coordinates
+    zs = np.zeros_like(ys)
+    zs[:, 0] = ys[:, 0]
+    zs[:, 1:] = (ys[:, 1:] - lower_bounds[1:]) / \
+                y_ranges[1:] * y_ranges[0] + lower_bounds[0]
+
+    #copy the axis for the other bcs
+    axes = [host] + [host.twinx() for i in range(ys.shape[1] - 1)]
+    for i, axis in enumerate(axes):
+        axis.set_ylim(lower_bounds[i], upper_bounds[i])
```

Comment:
I am a bit confused. If we transform `y` to be in a certain range above, but we then set the proper range here, doesn't the transformed `y` show up a bit wrong when we plot it?